### PR TITLE
Revise validation tests for depth/stencil formats for copies

### PR DESCRIPTION
**Pull request by Richard-Yunchao**
If the copied format is depth/stencil format, it requires that:
  - copy size must exactly equals to copied subresource's size
  - offset in buffer must be a multiple of 4

<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

**Comment by Richard-Yunchao** [comment]
Many depth/stencil formats have not been implemented in Chromium, but the copy related validation tests for depth/stencil with or without this change can be tested in latest Chromium via depth32float for T2B copy, which has already been there in Chromium.

File: src/webgpu/api/validation/image_copy/layout_related.spec.ts
Comment (kainino0x) [inline src/webgpu/api/validation/image_copy/layout_related.spec.ts:123]: This comment doesn't quite follow. Need to explain the middle step.
```suggestion
      // The test texture size will be rounded up from the copy size to the next valid texture size.
      // If the format is a depth/stencil format, its copy size must equal to subresource's size.
      // So filter out depth/stencil cases where the rounded-up texture size would be different from the copy size.
```

**Review by kainino0x — APPROVED** [review]
lgtm, thanks for fixing these! Are these tests currently marked as failing or did we somehow not notice they were broken?

**Comment by Richard-Yunchao** [comment]
> lgtm, thanks for fixing these! Are these tests currently marked as failing or did we somehow not notice they were broken?

They are not marked as failing. However, it is not a problem. All depth/stencil formats were thought non-copyable until PR https://github.com/gpuweb/cts/pull/691 (just merged yesterday), so all these failings didn't exist before that. After PR 691, we do need to update WebGPUExpectation in Chromium when we roll 691 (and this one) into Chromium because most depth/stencil formats have not implemented yet.

**Comment by Richard-Yunchao** [comment]
I guess it is OK to merge this change? This is not a huge PR. I don't think we need a second review.

**Comment by kainino0x** [comment]
Aha, got it, I see why this wasn't a problem before.

We normally do only one review on CTS changes.

**Comment by Jiawei-Shao** [comment]
> > lgtm, thanks for fixing these! Are these tests currently marked as failing or did we somehow not notice they were broken?
> 
> They are not marked as failing. However, it is not a problem. All depth/stencil formats were thought non-copyable in validation tests until PR #691 (just merged yesterday), so all these failings didn't exist before that. After PR 691, we do need to update WebGPUExpectation in Chromium when we roll 691 (and this one) into Chromium because most depth/stencil formats have not been implemented yet.
> 
> Currently, I just added SizedDepthStencilFormats into copy related validation because unsized depth/stencil formats are complicated (bytesPerBlock should be per aspect, not per format.), so any copy related validation tests for `depth16unorm` and `stencil8` may need to be marked as failing. I noticed that some folks are working on these formats in Dawn, though.

Hi @kainino0x  and @Richard-Yunchao,  now that the validation tests about depth/stencil copies are also added [here](https://github.com/gpuweb/cts/blob/main/src/webgpu/api/validation/encoding/cmds/buffer_texture_copies.spec.ts#L230), what's your plan to handle that file?

**Comment by Richard-Yunchao** [comment]
> > > lgtm, thanks for fixing these! Are these tests currently marked as failing or did we somehow not notice they were broken?
> > 
> > 
> > They are not marked as failing. However, it is not a problem. All depth/stencil formats were thought non-copyable in validation tests until PR #691 (just merged yesterday), so all these failings didn't exist before that. After PR 691, we do need to update WebGPUExpectation in Chromium when we roll 691 (and this one) into Chromium because most depth/stencil formats have not been implemented yet.
> > Currently, I just added SizedDepthStencilFormats into copy related validation because unsized depth/stencil formats are complicated (bytesPerBlock should be per aspect, not per format.), so any copy related validation tests for `depth16unorm` and `stencil8` may need to be marked as failing. I noticed that some folks are working on these formats in Dawn, though.
> 
> Hi @kainino0x and @Richard-Yunchao, now that the validation tests about depth/stencil copies are also added [here](https://github.com/gpuweb/cts/blob/main/src/webgpu/api/validation/encoding/cmds/buffer_texture_copies.spec.ts#L230), what's your plan to handle that file?

I will do a code refactoring between these two file folders. I will put B2T, T2B and writeTexture related tests in image_copy, and put B2B and T2T copies in encoding/cmds. WDYT? @kainino0x

**Comment by kainino0x** [comment]
SGTM, I think that's what the division was intended to be.

**Pull request merged**
